lwcomponent/component_internal_test.go

```diff
@@ -0,0 +1,223 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2020, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// A Lacework component package to help facilitate the loading and execution of components
+package lwcomponent
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	mockComponent = Component{
+		Name:        "lacework-mock-component",
+		Description: "This is a mock mock component",
+		Version:     "0.1.0",
+		Status:      "installed",
+		Signature:   "0df2d5957dd7583361dcc3a888b2ad9e3fa29a413bbf711a572f65348227d898",
+		CLICommand:  false,
+		CommandName: "",
+		Binary:      true,
+		Library:     false,
+		Standalone:  false,
+	}
+)
+
+func ensureMockComponent() (string, error) {
+	componentData := `#!/bin/bash
+echo "Hello World!"
+`
```

Comment:
we have now tests on windows systems so maybe a go tag to avoid running this on windows

Reply:
I'm assuming windows pipeline doesn't run as part of PR's since this is green?  I filed https://lacework.atlassian.net/browse/ALLY-843 to ensure component functionality on windows...